### PR TITLE
[SPARK-29648][SQL][TESTS] Port limit.sql

### PR DESCRIPTION
**Pull request by maropu**
### What changes were proposed in this pull request?

This PR ports limit.sql from PostgreSQL regression tests https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/sql/limit.sql

The expected results can be found in the link: https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/expected/limit.out

### Why are the changes needed?

To check behaviour differences between Spark and PostgreSQL

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

Pass the Jenkins. And, Comparison with PgSQL results

### CODE REVIEW

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    (123, 4567890123456789),
+    (4567890123456789, 123),
+    (4567890123456789, 4567890123456789),
+    (4567890123456789, -4567890123456789))
+  AS v(q1, q2);
+
+-- Test null limit and offset.  The planner would discard a simple null
+-- constant, so to ensure executor is exercised, do this:
+-- [SPARK-29650] Discard a NULL constant in LIMIT
+select * from int8_tbl limit (case when random() < 0.5 then bigint(null) end);
```
Comment (maropu) [inline sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql]: Filed: https://issues.apache.org/jira/browse/SPARK-29650

**Comment by maropu** [comment]
retest this please

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+-- [SPARK-27767] Built-in function: generate_series
+-- select generate_series(0,2) as s1, generate_series((random()*.1)::int,2) as s2
+-- order by s2 desc;
+
+-- test for failure to set all aggregates' aggtranstype
+-- explain (verbose, costs off)
+-- select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
+--   from tenk1 group by thousand order by thousand limit 3;
+
+select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql]: Although I believe `Apache Spark` behavior `random()*0` == `0.0` is more natural, do we have a JIRA issue for this difference from PostgreSQL where `random()*0` == `0`?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except one minor request to add a JIRA ID comment for the difference).

File: sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql (hunk 2 above)
Comment (maropu) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql]: Sorry, but I missed your point and the result seems to be the same;
```
select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
  from tenk1 group by thousand order by thousand limit 3;
  s1   |  s2   
-------+-------
 45000 | 45000
 45010 | 45010
 45020 | 45020
(3 rows)
```
https://github.com/postgres/postgres/blob/REL_12_STABLE/src/test/regress/expected/limit.out#L497-L504

Also, I tried below;
```
postgres=# select * from (values (1)) t(v) where random() * 0.0 = 0.0;
 v 
---
 1
(1 row)

scala> sql("""select * from (values (1)) t(v) where random() * 0.0 = 0.0""").show()
+---+
|  v|
+---+
|  1|
+---+
```
What's the difference here?

Comment (dongjoon-hyun) [reply sql/core/src/test/resources/sql-tests/inputs/postgreSQL/limit.sql]: The generated result is not the same in this PR.

File: sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+-- !query 6
+select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
+  from tenk1 group by thousand order by thousand limit 3
+-- !query 6 schema
+struct<s1:bigint,s2:double>
+-- !query 6 output
+45000	45000.0
+45010	45010.0
+45020	45020.0
```
Comment (dongjoon-hyun) [inline sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out]: The following is the original, @maropu .
```
select sum(tenthous) as s1, sum(tenthous) + random()*0 as s2
  from tenk1 group by thousand order by thousand limit 3;
  s1   |  s2   
-------+-------
 45000 | 45000
 45010 | 45010
 45020 | 45020
(3 rows)
```

Comment (maropu) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out]: You meant `45000` is different from `45000.0`, right (I might misunderstand your point though)? In pgSQL, `.0` is trimmed in printing;
```
postgres=# create table t(d float8);
CREATE TABLE
postgres=# insert into t values (3.0);
INSERT 0 1
postgres=# select * from t;
 d 
---
 3
(1 row)
```

Comment (dongjoon-hyun) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out]: Got it. My bad. Thank you for the explanation. I was confused.

**Comment by dongjoon-hyun** [comment]
Then, merged to master! Thank you, @maropu !

File: sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out (hunk 3 above)
Comment (maropu) [reply sql/core/src/test/resources/sql-tests/results/postgreSQL/limit.sql.out]: Yea, thanks for your check~

**Pull request closed without merging**